src/keys/spend/public.rs

```diff
@@ -0,0 +1,112 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::{
+    permutation, JubJubAffine, JubJubExtended, JubJubScalar, PublicKey,
+    StealthAddress,
+};
+
+use super::secret::SecretSpendKey;
+
+#[cfg(feature = "canon")]
+use canonical::Canon;
+#[cfg(feature = "canon")]
+use canonical_derive::Canon;
+
+use dusk_bytes::{DeserializableSlice, Error, HexDebug, Serializable};
+use dusk_jubjub::GENERATOR_EXTENDED;
+use subtle::{Choice, ConstantTimeEq};
+
+/// Public pair of `a·G` and `b·G` defining a [`PublicSpendKey`]
+#[derive(HexDebug, Clone, Copy)]
```

Comment:
We should implement `PartialEq` and `Eq` since there is no restriction for that and it will help to evaluate the correct and expected keys